core/src/main/groovy/noe/rhel/server/workspace/WorkspaceHttpdTomcatsBaseOS.groovy

```diff
@@ -0,0 +1,84 @@
+package noe.rhel.server.workspace
+
+import groovy.util.logging.Slf4j
+import noe.common.DefaultProperties
+import noe.common.utils.FileStateVault
+import noe.common.utils.JBFile
+import noe.common.utils.Version
+import noe.ews.server.tomcat.TomcatProperties
+import noe.rhel.server.httpd.HttpdBaseOS
+import noe.server.Httpd
+import noe.workspace.WorkspaceHttpdBaseOS
+import noe.workspace.WorkspaceMultipleTomcats
+import noe.workspace.WorkspaceTomcat
+
+@Slf4j
+class WorkspaceHttpdTomcatsBaseOS extends WorkspaceMultipleTomcats{
+
+  int numberOfAdditionalTomcats
+
+  protected static WorkspaceHttpdBaseOS workspaceHttpd
+  protected static WorkspaceTomcat workspaceTomcat
+  private FileStateVault moduleJkStateVault
+
+  Boolean ews
+  Boolean rpm
+  Version ewsVersion
+  String hostIpAddress = DefaultProperties.HOST
+  Map<String,String> originalTomcatHosts = [:]
+
+  WorkspaceHttpdTomcatsBaseOS(int numberOfAdditionalTomcats = 0) {
+    super()
+    this.numberOfAdditionalTomcats = numberOfAdditionalTomcats
+    //downloadClusterBench()
+    workspaceHttpd = new WorkspaceHttpdBaseOS()
+    workspaceTomcat = new WorkspaceTomcat()
+
+    this.ewsVersion = DefaultProperties.ewsVersion()
+  }
+
+  def prepare() {
+    log.debug("Creating workspace Httpd with ${numberOfAdditionalTomcats} Tomcats BaseOS.")
+    super.prepare()
+    workspaceHttpd.prepare()
+
+    HttpdBaseOS httpd = (HttpdBaseOS) serverController.getServerById(serverController.getHttpdServerId())
+    httpd.setHost(hostIpAddress)
+    httpd.updateConfSetBindAddress(hostIpAddress)
+
+    copyModulesIfMissing(httpd)
+    workspaceTomcat.prepare(true) //always true due to running with JWS 5 `ewsVersion >= new Version('3.1.0-DR0')`
+
+    createAdditionalTomcatsWithRegistrations(numberOfAdditionalTomcats)
+    originalTomcatHosts.putAll(originalTomcatHostsIpAddresses())
+
+    serverController.backup()
+  }
+
+  private copyModulesIfMissing(Httpd httpd) {
+    File modulePath = new File("${httpd.getServerRoot()}/modules/mod_jk.so")
+    File sclModulePath = new File("${DefaultProperties.HTTPD_SCL_ROOT}/usr/lib64/httpd/modules/mod_jk.so")
+    log.info("Storing content of file ${modulePath}.")
+    moduleJkStateVault = new FileStateVault().push(modulePath)
```

Comment:
What is the point of `FileStateVault`? IIRC you just copy mod_jk module, why not delete it simply?

Reply:
I can just remove it only if it is RHEL8 and JBCS mod_jk is copied, otherwise it has to stay as it is. So `FileStateVault` is best option. :)

Reply:
To me, the better option would be to create File instance only when copy is performed actually, and based on it's initialization I would delete the file - to prevent loading binaries to memory.

I think this is not good practice, because its over-using can lead to problems with memory requirements in a future. I would prefer more lightweight solution as proposed and use vaults for configurations files or logs mainly - not for binaries.

Reply:
Modified and squashed. Also I run tests and they pass.

Reply:
I don't see any relevant change tbh.

Reply:
Point was not use `FileStateVault` at all.

Reply:
The message from `FileStateVault` is quite clear https://github.com/web-servers/noe-core/blob/master/core/src/main/groovy/noe/common/utils/FileStateVault.groovy#L4 If there are some limitations or desired use cases then they should be mentioned. Otherwise it is not clear. 

Also if it is only meant for using with config files, then the better name should be `ConfigFileStateVaul`, otherwise it implies to use whatever it is suitable. I just want to use the best tool serving my needs.

Reply:
Ok, in that javadoc should be information that the content is stored to memory and should be used with care -> mass/over using can lead to OOM during runtime. 

Config file was example, so proposed renaming would be confusing as well.

Reply:
Solved in new commit `removing mod_jk.so files in RHEL8 JBCS`. Tested.